validation/src/main/java/io/micronaut/validation/validator/DefaultValidator.java

```diff
@@ -2074,17 +1798,51 @@ public <T extends Path.Node> T as(Class<T> nodeType) {
     }
 
     /**
-     * Method node implementation.
+     * Default constructor node.
      */
-    private final class DefaultParameterNode extends DefaultPropertyNode implements Path.ParameterNode {
+    private static final class DefaultConstructorNode extends DefaultMethodNode implements Path.ConstructorNode {
```

Comment:
make this a `record`

Reply:
Noticed that the code size could be reduced by using a common parent, so I created an abstract `DefaultNode` with 5 methods. `record` didn't seem to reduce the implementation, as it doesn't define getters in the desired format.